Assets/Scripts/IObstacles.cs

```diff
@@ -0,0 +1,6 @@
+
+public interface IObstacles
```

Comment:
nit: As it's an interface for a single obstacle, maybe rename it to `IObstacle`?

Reply:
like you mentioned to me about the naming convention,  i have now named it as "ObstacleController".

Reply:
and removed IObstacle interface class.